pkg/ruler/api.go

```diff
@@ -46,11 +46,12 @@ type AlertDiscovery struct {
 
 // Alert has info for an alert.
 type Alert struct {
-	Labels      labels.Labels `json:"labels"`
-	Annotations labels.Labels `json:"annotations"`
-	State       string        `json:"state"`
-	ActiveAt    *time.Time    `json:"activeAt"`
-	Value       string        `json:"value"`
+	Labels          labels.Labels `json:"labels"`
+	Annotations     labels.Labels `json:"annotations"`
+	State           string        `json:"state"`
+	ActiveAt        *time.Time    `json:"activeAt"`
+	KeepFiringSince *time.Time    `json:"keepFiringSince"`
```

Comment:
To match prometheus I think this needs to be `json:"keepFiringSince,omitempty`

Reply:
Yes, thanks for catching this. Updated.

I also verified the API response matches prometheus.

keepFiringSince is only included in the response when expected:

Response when alert is not resolved:
```
"groups": [
      {
        "name": "recording_rule",
        "file": "recording-rule",
        "rules": [
          {
            "name": "recording_rule",
            "query": "time() % 3600",
            "labels": {},
            "health": "ok",
            "lastError": "",
            "type": "recording",
            "lastEvaluation": "2024-03-21T08:26:52.745999513Z",
            "evaluationTime": 0.001055442
          }
        ],
        "interval": 60,
        "lastEvaluation": "2024-03-21T08:26:52.745964572Z",
        "evaluationTime": 0.001094416,
        "limit": 0
      },
      ...] 
``` 
Response when alert is resolved and keep firing for value is being used:

```
"groups": [
      {
        "name": "keep-firing2",
        "file": "rg-keep-firing",
        "rules": [
          {
            "state": "firing",
            "name": "alerting_rule_keepfiring60mins2",
            "query": "absent(recording_rule)",
            "duration": 0,
            "keepFiringFor": 3600,
            "labels": {},
            "annotations": {},
            "alerts": [
              {
                "labels": {
                  "alertname": "alerting_rule_keepfiring60mins2"
                },
                "annotations": {},
                "state": "firing",
                "activeAt": "2024-03-21T08:22:03.514363333Z",
                "keepFiringSince": "2024-03-21T08:27:03.514363333Z",
                "value": "1e+00"
              }
            ],
            "health": "ok",
            "lastError": "",
            "type": "alerting",
            "lastEvaluation": "2024-03-21T08:29:03.515574705Z",
            "evaluationTime": 0.002663176
          },
          ...]
```